.github/workflows/e2e.yml

```diff
@@ -15,75 +15,86 @@
       - uses: actions/checkout@v2.3.4
         with:
           path: plugin
+          ref: ${{ github.event.workflow_run.head_sha }}
+
       - name: Checkout e2e test repo
         uses: actions/checkout@v2.3.4
         with:
           repository: cdapio/cdap-e2e-tests
           path: e2e
+
       - name: Cache
         uses: actions/cache@v2.1.3
         with:
           path: ~/.m2/repository
           key: ${{ runner.os }}-maven-${{ github.workflow }}-${{ hashFiles('**/pom.xml') }}
           restore-keys: |
             ${{ runner.os }}-maven-${{ github.workflow }}
+
+      - name: Get Secrets from GCP Secret Manager
+        id: 'secrets'
+        uses: 'google-github-actions/get-secretmanager-secrets@v0'
+        with:
+          secrets: |-
+            SALESFORCE_USERNAME:cdapio-github-builds/SALESFORCE_USERNAME
+            SALESFORCE_PASSWORD:cdapio-github-builds/SALESFORCE_PASSWORD
+            SALESFORCE_SECURITY_TOKEN:cdapio-github-builds/SALESFORCE_SECURITY_TOKEN
+            SALESFORCE_CONSUMER_KEY:cdapio-github-builds/SALESFORCE_CONSUMER_KEY
+            SALESFORCE_CONSUMER_SECRET:cdapio-github-builds/SALESFORCE_CONSUMER_SECRET
+
       - name: Run tests
         run: python3 e2e/src/main/scripts/run_e2e_test.py
-      - name: Upload report 
+        env:
+          SALESFORCE_USERNAME: ${{ steps.secrets.outputs.SALESFORCE_USERNAME }}
+          SALESFORCE_PASSWORD: ${{ steps.secrets.outputs.SALESFORCE_PASSWORD }}
+          SALESFORCE_SECURITY_TOKEN: ${{ steps.secrets.outputs.SALESFORCE_SECURITY_TOKEN }}
+          SALESFORCE_CONSUMER_KEY: ${{ steps.secrets.outputs.SALESFORCE_CONSUMER_KEY }}
+          SALESFORCE_CONSUMER_SECRET: ${{ steps.secrets.outputs.SALESFORCE_CONSUMER_SECRET }}
+
+      - name: Upload report
         uses: actions/upload-artifact@v2.2.4
         if: always()
         with:
           name: Cucumber report
           path: ./plugin/target/cucumber-reports
-      - name: Upload debug files 
+
+      - name: Upload debug files
         uses: actions/upload-artifact@v2.2.4
         if: always()
         with:
           name: Debug files
           path: ./**/target/e2e-debug
-      - name: Deploy report
-        uses: peaceiris/actions-gh-pages@v3
+
+      - name: Upload reports to GCS
+        uses: google-github-actions/upload-cloud-storage@v0
         if: always()
         with:
-          github_token: ${{ secrets.GITHUB_TOKEN }}
-          publish_dir: ./plugin/target/cucumber-reports
-          destination_dir: cucumber-reports/${{ github.ref }}
-          
+          path: ./plugin/target/cucumber-reports
+          destination: e2e-tests-cucumber-reports/${{ github.event.repository.name }}/${{ github.ref }}
+
       - name: github-status-action
         uses: Sibz/github-status-action@67af1f4042a5a790681aad83c44008ca6cfab83d
         if: always()
         with:
-          authToken: ${{ secrets.GITHUB_TOKEN }}
+          authToken: ${{ secrets.AUTH_TOKEN }}
```

Comment:
why is this change needed?

Reply:
reverted the change (18bd6bbba7443d9e5c842d005d05000f3f295928)